src/carousel/carousel.ts

```diff
@@ -280,7 +280,30 @@ export class NgbCarousel implements AfterContentChecked,
       });
     }
 
-    this.slides.changes.pipe(takeUntil(this._destroy$)).subscribe(() => this._cd.markForCheck());
+    this.slides.changes.pipe(takeUntil(this._destroy$)).subscribe(() => {
+      const transitionIds = this._transitionIds;
+      if (transitionIds) {
+        for (const id of transitionIds) {
+          ngbCompleteTransition(this._getSlideElement(id));
+        }
+        this._transitionIds = null;
+      }
+
+      this._cd.markForCheck();
+
+      this._ngZone.onStable.pipe(take(1)).subscribe(() => {
+        const activeId = this.activeId;
```

Comment:
nitpick: what's the point of having `const activeId` and `const id`?

Shorter version:
```ts
for (const {id} of this.slides) {
  const element = this._getSlideElement(id);
  if (id === this.activeId) {
    element.classList.add('active');
  } else {
    element.classList.remove('active');
  }
}
```